README.md

```diff
@@ -1,29 +1,190 @@
+| Linux                                    | *Host* *                                                                                             | Ubuntu 18.04/20.04   | `host`                       |
 
-Support scripts for creating Micro-ROS agent and client workspaces,
-and cleanly doing cross-compilation.
+Please note that NuttX with Olimex STM32-E407 board is the reference platform and not everything might be supported on other platforms.
 
-See [micro_ros_setup README](micro_ros_setup/README.md) for usage instructions.
+*\* Support for compiling apps in a native Linux host for testing and debugging*
 
-## Purpose of the project
+# Dependencies
 
-The software is not ready for production use. It has neither been developed nor tested for a specific use case. However, the license conditions of the applicable Open Source licenses allow you to adapt the software to your needs. Before using it in a safety relevant setting, make sure that the software fulfills your requirements and adjust it according to any applicable safety standards (e.g. ISO 26262).
+This package targets **ROS 2** installation. ROS 2 supported distributions are:
+
+| ROS 2 Distro | State     | Branch    |
+| ------------ | --------- | --------- |
+| Crystal      | Supported | `crystal` |
+| Dashing      | Supported | `dashing` |
+| Foxy         | Supported | `foxy`    |
+
+Some other prerequisites needed for building a firmware using this package are:
+
+```
+sudo apt install python3-rosdep
+```
+
+# Building 
+
+Create a ROS 2 workspace and build this package for a given ROS 2 distro (see table above):
+
+```bash
+source /opt/ros/$ROS_DISTRO/setup.bash
+
+mkdir uros_ws && cd uros_ws
+
+git clone -b $ROS_DISTRO https://github.com/micro-ROS/micro-ros-build.git src/micro-ros-build
+
+rosdep update && rosdep install --from-path src --ignore-src -y
+
+colcon build
+
+source install/local_setup.bash
+```
+
+Once the package is built, the firmware scripts are ready to run.
+
+You can find first steps tutorials in [micro-ROS webpage](https://micro-ros.github.io/docs/tutorials/core/first_application_rtos/).
+
+
+## Creating micro-ROS firmware
+
+Using `create_firmware_ws.sh [RTOS] [Platform]` command a firmware folder will be created with the required code for building a micro-ROS app. For example, for our reference platform, the invocation is:
+
+```bash
+# Creating a NuttX + micro-ROS firmware workspace
+ros2 run micro_ros_setup create_firmware_ws.sh nuttx olimex-stm32-e407
+
+# Creating a FreeRTOS + micro-ROS firmware workspace
+ros2 run micro_ros_setup create_firmware_ws.sh freertos olimex-stm32-e407
+
+# Creating a Zephyr + micro-ROS firmware workspace
+ros2 run micro_ros_setup create_firmware_ws.sh zephyr olimex-stm32-e407
+```
+
+## Configuring micro-ROS firmware
+
+By running `configure_firmware.sh` command the installed firmware is configured and modified in a pre-build step. Usually this command will show its usage if parameters are required:
```

Comment:
```suggestion
By running the `configure_firmware.sh` command, the installed firmware is configured and modified in a pre-build step. Usually this command will show its usage if parameters are required:
```

Reply:
It's not really clear to me what the sentence "Usually this command will show its usage if parameters are required:" means

Reply:
Modified